compiler/noirc_frontend/src/lexer/lexer.rs

```diff
@@ -424,10 +433,20 @@ impl<'a> Lexer<'a> {
 impl<'a> Iterator for Lexer<'a> {
     type Item = SpannedTokenResult;
     fn next(&mut self) -> Option<Self::Item> {
-        if self.done {
-            None
-        } else {
-            Some(self.next_token())
+        loop {
+            if self.done {
+                return None;
+            } else {
+                let token = self.next_token();
+
+                if self.skip_trivia
+                    && token.as_ref().map_or(false, |spanned| spanned.token().is_trivia())
+                {
+                    continue;
+                }
```

Comment:
Instead of having to check every single token, I think we should remove this and only check before issuing a comment token.